src/buildblock/DynamicProjData.cxx

```diff
@@ -162,10 +163,20 @@ read_from_file(const string& filename) // The written projection data is read in
     }
 #endif // end of HAVE_LLN_MATRIX
 
+  if (strncmp(signature, "Multi", 5) == 0) {
+
+#ifndef NDEBUG
+        info(boost::format("DynamicProjData::read_from_file trying to read %s as a Multi file.") % filename);
+#endif
+
+      shared_ptr<DynamicProjData> dynamic_proj_data(new DynamicProjData(*MultipleProjData::read_from_file(filename)));
```

Comment:
this might be a memory leak, but in any case seems superfluous. why not simply
```C++
 shared_ptr<DynamicProjData> dynamic_proj_data(MultipleProjData::read_from_file(filename));
```

Reply:
Your suggestion gives the following error:
```
/usr/local/include/boost/smart_ptr/shared_ptr.hpp:341: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'shared_ptr<stir::MultipleProjData>' to 'const shared_ptr<stir::DynamicProjData>' for 1st argument
```

Reply:
ah right. it won't call the constructor to convert `MultipleProjData` to `DynamicProjData`.
Then something ugly like
```
   unique_ptr<MultipeProjData> multi_proj_data(MultipleProjData::read_from_file(filename)));
  shared_ptr<DynamicProjData> dynamic_proj_data = MAKE_SHARED<DynamicProjData>(*multi_proj_data));
```
will at least not have the leak. Would be better to template `read_from_file` to avoid the copy, but fine. (if we do that though, can we use it for images than as well?)